### PR TITLE
Resolve #1: Where to directly return groups

### LINKED ISSUE #1: Query::where() should directly return Group[]
# Problem

Currently `Query` has the following structure:

```php
$where = $query->where();
if ($where !== null) {
    foreach ($where->groups() as $group) {
        foreach ($group->conditions() as $condition) {
            //
        }
    }
}
$orders = $query->orders();
foreach ($orders as $order) {
    //
}
```

However, they seems to be weird and inconsistent.

# Proposal

## A

```php
$where = $query->where();
if ($where !== null) {
    foreach ($where->groups() as $group) {
        foreach ($group->conditions() as $condition) {
            //
        }
    }
}
$orderBy = $query->orderBy();
if ($orderBy !== null) {
    foreach ($orderBy->orders() as $order) {
        //
    }
}
```

## B

```php
foreach ($query->where() as $group) {
    foreach ($group->conditions() as $condition) {
        //
    }
}
foreach ($query->orders() as $order) {
    //
}
```

Which is better?

( Note: Each component implements `\IteratorAggregate` so we can directly traverse it without invoking method )

### PR DESCRIPTION
**Pull request by mpyw**
- `Conditions\Condition` is replaced with `Condition`
- `Conditions\Group` is replaced with `ConditionGroup`
- `Where` is replaced with `ConditionGroup[]`
- `Where::create()` is replaced with `ConditionGroup::createMany()`
- <del>`Select::conditionGroups()` is added as an alias of `Select::where()`</del>

### CODE REVIEW

File: src/Query/Select.php [hunk 1]
```diff
@@ -46,14 +46,24 @@ protected static function validate(Order ...$orders)
...
     {
         return $this->where;
     }
 
     /**
+     * An alias of where().
+     *
+     * @return ConditionGroup[]
+     */
+    public function conditionGroups()
```
Comment (chitoku-k) [inline src/Query/Select.php]: Why aliasing?

Comment (mpyw) [reply src/Query/Select.php]: We can't easily assume that `where()` returns an array. Longer but clearer alias may be useful for newcomer.

Comment (chitoku-k) [reply src/Query/Select.php]: Let's say a newcomer looks at a list of completions. These duplicate entries look like there are certain differences between the methods: the same return-type but different names. I am opposed to this unnecessary alias.

Comment (mpyw) [reply src/Query/Select.php]: Okay

**Pull request merged**